File: 0026-remove-duplicates-from-sorted-array/26.cpp
/*
 * @lc app=leetcode id=26 lang=cpp
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
const static auto fast = [] {
  ios_base::sync_with_stdio(false);
  cin.tie(nullptr);
  cout.tie(nullptr);
  return 0;
}();

class Solution {
public:
  int removeDuplicates(vector<int>& nums) {
    /** Approach: Two Pointers
     */
    int l   = 1;
    int pos = 1;
    int n   = nums.size();

    while (l < n) {
      if (nums[l] != nums[pos - 1]) nums[pos++] = nums[l];
      l++;
    }
    return pos;
  }
};
// @lc code=end

File: 0026-remove-duplicates-from-sorted-array/26.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let mut pos = 1;
        for i in 1..nums.len() {
            if nums[i] != nums[pos - 1] {
                nums[pos] = nums[i];
                pos += 1;
            }
        }
        pos as i32
    }
}
